util/feeding.py

```diff
@@ -109,14 +109,15 @@ def to_sparse_tuple(sequence):
 
 def create_dataset(sources,
                    batch_size,
+                   limit=-1,
                    enable_cache=False,
                    cache_path=None,
                    train_phase=False,
                    exception_box=None,
                    process_ahead=None,
                    buffering=1 * MEGABYTE):
     def generate_values():
-        samples = samples_from_files(sources, buffering=buffering, labeled=True)
+        samples = samples_from_files(sources, buffering=buffering, labeled=True, sample_limit=limit)
```

Comment:
If we won't go for the above proposal: Why not applying the limit within the following loop (exiting it once the limit is reached)?

Yes, this has the disadvantage of still having to load the complete "lists" of the sample sources (SDB offset table, CSV rows), but on the other hand it would be:
- less intrusive and therefore easier to maintain and 
- far more logical in regards to the outcome: Limiting the combination of all samples instead of limiting each source

Reply:
> far more logical in regards to the outcome: Limiting the combination of all samples instead of limiting each source

You suggested the syntax limiting per-files, so I'm lost here.



> less intrusive and therefore easier to maintain and

Much much painful to test, though, because `create_dataset` produces tensorflow data ops.

Reply:
https://github.com/mozilla/DeepSpeech/issues/2777#issuecomment-603392850

Reply:
FWIW for testing a dataset, it shouldn't be that hard. In eager mode, just `list(dataset)` will do what you expect. Example: https://gist.github.com/reuben/81ac06018a6bcc581a48d08c51a01a5b

Reply:
> FWIW for testing a dataset, it shouldn't be that hard. In eager mode, just `list(dataset)` will do what you expect. Example: https://gist.github.com/reuben/81ac06018a6bcc581a48d08c51a01a5b

Yes, but that's starting to be quite a lot of setup for unit testing :/